src/lib.rs

```diff
@@ -310,3 +318,238 @@ pub fn platform() -> Platform {
+            "armv6" => Ok(Arch::ArmV6),
+            "armv7" => Ok(Arch::ArmV7),
+            "asmjs" => Ok(Arch::AsmJs),
+            "hexagon" => Ok(Arch::Hexagon),
+            "i386" => Ok(Arch::I386),
+            "i586" => Ok(Arch::I586),
+            "i686" | "i686-AT386" => Ok(Arch::I686),
+            "m68k" => Ok(Arch::M68k),
+            "mips" => Ok(Arch::Mips),
+            "mipsel" => Ok(Arch::MipsEl),
+            "mips64" => Ok(Arch::Mips64),
+            "mips64el" => Ok(Arch::Mips64El),
+            "powerpc" | "ppc" | "ppcle" => {
+                Ok(Arch::PowerPc)
+            }
+            "powerpc64" | "ppc64" | "ppc64le" => Ok(Arch::PowerPc64),
+            "powerpc64le" => Ok(Arch::PowerPc64Le),
+            "riscv32gc" => Ok(Arch::Riscv32Gc),
+            "riscv64gc" | "riscv64" => Ok(Arch::Riscv64Gc),
+            "s390x" => Ok(Arch::S390x),
+            "sparc" => Ok(Arch::Sparc),
+            "sparc64" => Ok(Arch::Sparc64),
+            "thumbv7neon" => Ok(Arch::ThumbV7Neon),
+            "wasm32" => Ok(Arch::Wasm32),
+            "wasm64" => Ok(Arch::Wasm64),
+            "x86_64" | "amd64" => Ok(Arch::X86_64),
+            _ => Ok(Arch::Unknown(s.to_owned())),
+        }
+    }
+}
+
+impl std::fmt::Display for Arch {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if let Arch::Unknown(_) = self {
+            write!(f, "Unknown: ")?;
+        }
+
+        write!(
+            f,
+            "{}",
+            match self {
+                Arch::Aarch64 => "ARM64",
+                Arch::Arm => "ARM",
+                Arch::ArmEb => "ARM BE8",
+                Arch::ArmV4T => "ARMv4T",
+                Arch::ArmV5Te => "ARMv5TE",
+                Arch::ArmV6 => "ARMv6",
+                Arch::ArmV7 => "ARMv7",
+                Arch::AsmJs => "asm.js",
+                Arch::Hexagon => "Hexagon",
+                Arch::I386 => "i386",
+                Arch::I586 => "i586",
+                Arch::I686 => "i686",
+                Arch::M68k => "m68k",
+                Arch::Mips => "MIPS",
+                Arch::MipsEl => "MIPS (LE)",
+                Arch::Mips64 => "MIPS64",
+                Arch::Mips64El => "MIPS64 (LE)",
+                Arch::MipsIsa32R6 => "MIPS ISA 32r6",
+                Arch::MipsIsa32R6El => "MIPS ISA 32r6el",
+                Arch::PowerPc => "PowerPC",
+                Arch::PowerPc64 => "PowerPC64",
+                Arch::PowerPc64Le => "PowerPC64LE",
+                Arch::Riscv32Gc => "RISC-V32",
+                Arch::Riscv64Gc => "RISC-V64",
+                Arch::S390x => "S390x",
+                Arch::Sparc => "SPARC",
+                Arch::Sparc64 => "SPARC64",
+                Arch::ThumbV7A => "Thumbv7a",
+                Arch::ThumbV7Neon => "Thumb2-mode ARMv7a with NEON",
+                Arch::Wasm32 => "32-bit Web Assembly",
+                Arch::Wasm64 => "64-bit Web Assembly",
+                Arch::X86_64 => "X86_64/Amd64",
```

Comment:
I think we might want this to print out the arch directly from the toolchain triple instead of a more human-readable form.